kotlin/src/main/java/app/rive/runtime/kotlin/RiveArtboardRenderer.kt

```diff
@@ -183,121 +217,147 @@ open class RiveArtboardRenderer(
+                    _stop(stateMachine)
+                }
+            } else {
+                _animations(animationNames).forEach { animation ->
+                    _stop(animation)
+                }
             }
         }
     }
 
 
     fun stopAnimations(animationName: String, isStateMachine: Boolean = false) {
-        if (isStateMachine) {
-            _stateMachines(animationName).forEach { stateMachine ->
-                _stop(stateMachine)
-            }
-        } else {
+        synchronized(lock) {
+            if (isStateMachine) {
+                _stateMachines(animationName).forEach { stateMachine ->
+                    _stop(stateMachine)
+                }
+            } else {
 
-            _animations(animationName).forEach { animation ->
-                _stop(animation)
+                _animations(animationName).forEach { animation ->
+                    _stop(animation)
+                }
             }
         }
     }
 
     fun fireState(stateMachineName: String, inputName: String) {
-        val stateMachineInstances = _getOrCreateStateMachines(stateMachineName)
-        stateMachineInstances.forEach {
-            (it.input(inputName) as SMITrigger).fire()
-            _play(it, settleStateMachineState = false)
+        synchronized(lock) {
+            val stateMachineInstances = _getOrCreateStateMachines(stateMachineName)
+            stateMachineInstances.forEach {
+                (it.input(inputName) as SMITrigger).fire()
+                _play(it, settleStateMachineState = false)
+            }
         }
     }
 
     fun setBooleanState(stateMachineName: String, inputName: String, value: Boolean) {
-        val stateMachineInstances = _getOrCreateStateMachines(stateMachineName)
-        stateMachineInstances.forEach {
-            (it.input(inputName) as SMIBoolean).value = value
-            _play(it, settleStateMachineState = false)
+        synchronized(lock) {
+            val stateMachineInstances = _getOrCreateStateMachines(stateMachineName)
+            stateMachineInstances.forEach {
+                (it.input(inputName) as SMIBoolean).value = value
+                _play(it, settleStateMachineState = false)
+            }
         }
     }
 
     fun setNumberState(stateMachineName: String, inputName: String, value: Float) {
-        val stateMachineInstances = _getOrCreateStateMachines(stateMachineName)
-        stateMachineInstances.forEach {
-            (it.input(inputName) as SMINumber).value = value
-            _play(it, settleStateMachineState = false)
+        synchronized(lock) {
+            val stateMachineInstances = _getOrCreateStateMachines(stateMachineName)
+            stateMachineInstances.forEach {
+                (it.input(inputName) as SMINumber).value = value
+                _play(it, settleStateMachineState = false)
+            }
         }
     }
 
-    // PRIVATE FUNCTIONS
+// PRIVATE FUNCTIONS
```

Comment:
Mmh, so we're `synchronize`-ing all our public APIs and leaving the private ones unflitered?
Our private functions below contain a few read/write ops that could be considered critical sections.
Shouldn't we `synchronize` those instead? Keep our critical sections minimal